examples/tutorial/hello/java/README.md

```diff
@@ -1,3 +1,188 @@
-# Fn: Java
+# Java Function Hello World
 
-To learn how to build Java functions check out the [Fn Java FDK](https://github.com/fnproject/fdk-java)
+This example shows you how to test and deploy Java code to Fn. It also demonstrates passing text data to your function through `stdin`. 
+
+To learn more about using Java with Fn check out the [Fn Java FDK](https://github.com/fnproject/fdk-java).
+
+## Before you Begin
+
+This tutorial assumes you have installed Docker, Fn server, and Fn CLI.
+
+## Start Fn Server
+
+Start up the Fn server so we can deploy our function.
+
+>```sh
+>fn start
+>```
+
+The command starts Fn in single server mode using an embedded database and message queue. You can find all the
+configuration options [here](../../../../docs/operating/options.md). 
+
+## Create your Function 
+
+1. Change into the directory where you want to create your function.
+1. Run the following command to create a boilerplate Java function: 
+
+>```sh
+>fn init --runtime java hello
+>```
+    
+>A directory named `hello` is created with several files and directories in it.
+
+<ol start="3">
+  <li>Open the generated <code>src/main/java/com/example/fn/HelloFunction.java</code> file and you will see the following source code.</li>
+</ol>
+
+```java
+package com.example.fn;
+
+public class HelloFunction {
+
+    public String handleRequest(String input) {
+        String name = (input == null || input.isEmpty()) ? "world"  : input;
+
+        return "Hello, " + name + "!";
+    }
+
+}
+```
+
+<ol start="4">
+  <li>The command also generates a <code>yaml</code> metadata file. Open the <code>func.yaml</code> file.</li>
+</ol>
+
+```yaml
+version: 0.0.1
+runtime: java
+cmd: com.example.fn.HelloFunction::handleRequest
+build_image: fnproject/fn-java-fdk-build:jdk9-1.0.56
+run_image: fnproject/fn-java-fdk:jdk9-1.0.56
+format: http
+```
+
+The generated `func.yaml` file contains metadata about your function and declares a number of properties including:
+
+* `version`: Automatically starting at 0.0.1.
+* `runtime`: Set to `java` from the command line.
+* `cmd`: Name of the method to invoke. In this case `handleRequest` of the `com.example.fn.HelloFunction` class.
+
+These fields are set by default when you run `init` on a function. For more details on [function files go here](../../../../docs/developers/function-file.md).
+
+## Add Fn Registry Environment Variable
+
+Before we start developing we need to set the `FN_REGISTRY` environment variable. Normally, set the value to your Docker Hub username. However, you can work with Fn locally.  Set the `FN_REGISTRY` variable to an invented value: `noreg`.
```

Comment:
We should say it's preferable to set FN_REGISTRY to your Docker Hub id but you can use an invented value if only working locally and not pushing to Docker Hub.